category/applicative.lean

```diff
@@ -0,0 +1,153 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Instances for identity and composition functors
+-/
+
+import category.functor
+
+universe variables u v w u' v' w'
+
+section lemmas
+
+open function applicative
+
+variables {α β γ : Type u}
+variables {f : Type u → Type v}
+variables [applicative f] [is_lawful_applicative f]
+variables (g : β → γ)
+
+lemma pure_seq_pure {α β : Type u} (g : α → β) (x : α) :
+  pure g <*> (pure x : f α) = pure (g x) :=
+by simp
+
+open function
+
+lemma applicative.map_seq {β γ σ : Type u} (h : γ → σ) (x : f (β → γ)) (y : f β) :
+  h <$> (x <*> y) = (comp h <$> x) <*> y :=
+by rw [← pure_seq_eq_map,← pure_seq_eq_map,
+       seq_assoc,map_pure]
+
+lemma applicative.seq_map {β γ σ : Type u} (h : σ → β) (x : f (β → γ)) (y : f σ) :
+  x <*> (h <$> y) = (flip comp h) <$> x <*> y :=
+begin
+  rw [← pure_seq_eq_map,← pure_seq_eq_map,
+      seq_assoc,
+      seq_pure,
+      pure_seq_eq_map,
+      ← comp_map] ,
+  refl
+end
```

Comment:
Style: no space before `,`, add a space after, and fill the line (word wrap) rather than one line per rewrite. Also, this could be written `by rw [...]; refl`.